index.go

```diff
@@ -142,6 +148,42 @@ func (idx *faissIndex) Search(x []float32, k int64) (
 	return
 }
 
+func (idx *faissIndex) SearchWithoutIDs(x []float32, k int64, exclude []int64) (
```

Comment:
Given that this is going to be a public API, let's handle the situation where `len(exclude) == 0` in which case we just forward the call to `idx.Search(..)`.

Reply:
Done.
Didn't do it earlier because I already preempted such a situation in zapx by checking if the number of excluded IDs are > 0.

Reply:
Updating zapx to remove that condition then, pls check the zapx PR out.